src/main/appender.cpp

```diff
@@ -375,6 +377,31 @@ void Appender::FlushInternal(ColumnDataCollection &collection) {
 	context->Append(*description, collection);
 }
 
+void Appender::AppendDefault() {
+	if (!defaults[column]) {
+		throw InvalidInputException("Failed to append DEFAULT, this column does not have a DEFAULT value");
+	}
+	auto &default_expr = *defaults[column];
+	if (!default_expr.IsScalar()) {
+		throw InvalidInputException("Only columns with simple DEFAULT values are supported");
+	}
+
+	auto default_copy = default_expr.Copy();
+
+	auto &type = types[column];
+	auto binder = Binder::CreateBinder(*context);
+	ConstantBinder default_binder(*binder, *context, "DEFAULT value");
+	default_binder.target_type = type;
+	auto bound_default = default_binder.Bind(default_copy);
+
+	Value result_value;
+	if (!ExpressionExecutor::TryEvaluateScalar(*context, *bound_default, result_value)) {
+		throw InvalidInputException("Could not execute the DEFAULT expression of this column, please insert manually");
```

Comment:
This is also more of an assertion, we checked `IsScalar` so this should never fail